app/models/merchant.rb

```diff
@@ -32,4 +32,11 @@ def self.revenue_date(merchant_id, date)
     .where("merchants.id = #{merchant_id}").where(invoices: {created_at: date})
     .sum("invoice_items.unit_price * invoice_items.quantity")
   end
+
+  def self.most_items(number)
+    Merchant.joins(invoices: [:transactions, :invoice_items])
```

Comment:
Interesting. I kind of like it because in a long AR query helps to know where we're starting even if it is in the model already